GlslSandbox class syntax

marioecg opened the pull request:
I'm making a PR to port `GlslSandbox` function to an ES6 class.

In `example.js` enable importing the `GlslSandbox` class and you'll see the canvas completely in black 😅. I'd love help to know where to start debugging this.

marioecg committed:
directly assign value to uniforms

marioecg committed:
npm install

marioecg committed:
GlslSandbox to class syntax

marioecg committed:
pass uniforms to functions fixes the black canvas

patriciogonzalezvivo commented on class.js:
```diff
@@ -0,0 +1,405 @@
+        // Buffers
+        for (let i = 0, il = this.buffers.length; i < il; i++) {
+            let b = this.buffers[i];
+            if (db.width <= 1.0 && db.height <= 1.0)
```
There is a fix that needs to be update here https://github.com/patriciogonzalezvivo/glsl-sandbox/commit/f3bfc7c2cd9a30df7da9672418847a4a02bc232d

patriciogonzalezvivo commented:
if works as expected, let's replace index for it. Also please make sure that changes made after you branch are considered.
Gracias Mario!

Commit:
missed semicolons

Commit:
simplify function

Commit:
replace file for class structure

marioecg committed:
Merge branch 'main' into glsl-sandbox-class

marioecg commented:
I replaced `index.js` with the class structure and it works like a charm. The issue we were having before was that the main uniforms object wasn't being added when creating new shader materials, so now they're passed through the `createShaderMaterial` function like so:

``let material = createShaderMaterial(`#define BUFFER_${index}\n${frag_src}`, this.uniforms);``

I also updated with the fix from above. Let me know what you think!

Commit:
make sure latest updates are in branch

patriciogonzalezvivo commented:
Fantastic job! Thank you!
